### PR TITLE
84763 Validating claim status api response for success & failure stat…

### PR DESCRIPTION
**Pull request by kanchanasuriya**
## Summary

- *Implemented claim status api response for success & failure status validation*
- *Currently, rspecs cover one of the success & failure status. Rspecs for validating other success & failure scenarios will be covered in seperate PR*

## Related issue(s)

- *https://app.zenhub.com/workspaces/check-in-experience-61fc23a2cb8a14001132e102/issues/gh/department-of-veterans-affairs/va.gov-team/84763*

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
NA

## Acceptance criteria

- [x]  I added unit tests and integration tests for each feature (if applicable).
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: modules/check_in/app/sidekiq/check_in/travel_claim_status_check_worker.rb [hunk 1]
```diff
@@ -46,36 +51,84 @@ def claim_status(opts = {})
 
     # rubocop:disable Metrics/MethodLength
     def handle_response(opts = {})
-      # claim_response_body = opts[:claim_status_resp].body || []
+      claim_response_body = opts[:claim_status_resp].body || []
       claim_response_status = opts[:claim_status_resp].status
       facility_type = opts[:facility_type] || ''
       claim_number = nil
 
       statsd_metric, template_id = case claim_response_status
```
Comment (randomsync) [inline modules/check_in/app/sidekiq/check_in/travel_claim_status_check_worker.rb:59]: I'd suggest to move this to a separate method.

File: modules/check_in/app/sidekiq/check_in/travel_claim_status_check_worker.rb [hunk 2]
```diff
@@ -46,36 +51,84 @@ def claim_status(opts = {})
 
     # rubocop:disable Metrics/MethodLength
```
Comment (stevenjcumming) [inline modules/check_in/app/sidekiq/check_in/travel_claim_status_check_worker.rb:52]: @kanchanasuriya Can you move the case statement to another method and resolve the Metrics/MethodLength check.

Comment (kanchanasuriya) [reply modules/check_in/app/sidekiq/check_in/travel_claim_status_check_worker.rb]: Please refer to the latest commit

File: modules/check_in/app/sidekiq/check_in/travel_claim_status_check_worker.rb (hunk 1 above)
Comment (kanchanasuriya) [reply modules/check_in/app/sidekiq/check_in/travel_claim_status_check_worker.rb:64]: Please refer to the latest commit

**Review by randomsync — APPROVED** [review]
LGTM

**Pull request merged**